addons/mass_mailing/controllers/main.py

```diff
@@ -185,7 +185,7 @@ def _prepare_mailing_subscription_values(self, mailing, document_id, email, hash
         ).list_id.filtered('active')
         lists_optout = contacts.subscription_ids.filtered(
             lambda sub: sub.opt_out and sub.list_id not in lists_optin
-        ).list_id.filtered('active')
+        ).list_id.filtered('active').filtered('is_public')
```

Comment:
Let's maybe not change this in stable as it seems `lists_optout` is used in multiple places + it might mess with any override.

Would be better to specifically hide these in the view e.g. via `t-if="list not in lists_optout or list.is_public"`